### PR TITLE
[libc++] Properly define _LIBCPP_HAS_NO_UNICODE in __config_site

### LINKED ISSUE #93638: [libc++] _LIBCPP_HAS_NO_UNICODE is missing from __config_site
`_LIBCPP_HAS_NO_UNICODE` is not defined in `__config_site.in` and causes the CMake variable `LIBCXX_ENABLE_UNICODE` to not be honored.

### PR DESCRIPTION
**Pull request by ldionne**
Fixes #93638

### CODE REVIEW

**Review by mordante — APPROVED** [review]
Thanks! LGTM!

**Comment by ldionne** [comment]
@mordante , I am getting CI failures in

```
libcxx/test/libcxx/clang_tidy.gen.py
libcxx/test/libcxx/input.output/iostream.format/output.streams/ostream.formatted/ostream.formatted.print/vprint_unicode.pass.cpp
libcxx/test/libcxx/input.output/iostream.format/print.fun/transcoding.pass.cpp
libcxx/test/libcxx/input.output/iostream.format/print.fun/vprint_unicode_posix.pass.cpp
libcxx/test/libcxx/input.output/iostream.format/print.fun/vprint_unicode_windows.pass.cpp
libcxx/test/libcxx/utilities/format/format.functions/ascii.pass.cpp
libcxx/test/libcxx/utilities/format/format.functions/escaped_output.ascii.pass.cpp
libcxx/test/libcxx/utilities/format/format.string/format.string.std/extended_grapheme_cluster.pass.cpp
libcxx/test/std/input.output/iostream.format/output.streams/ostream.formatted/ostream.formatted.print/locale-specific_form.pass.cpp
libcxx/test/std/input.output/iostream.format/output.streams/ostream.formatted/ostream.formatted.print/vprint_unicode.pass.cpp
libcxx/test/std/input.output/iostream.format/print.fun/no_file_description.pass.cpp
libcxx/test/std/input.output/iostream.format/print.fun/vprint_unicode.sh.cpp
libcxx/test/std/utilities/format/format.formatter/format.formatter.spec/formatter.char.fsigned-char.pass.cpp
libcxx/test/std/utilities/format/format.formatter/format.formatter.spec/formatter.char.funsigned-char.pass.cpp
```

However, I am really not certain how to best fix those in most cases. Since this is more your cup of tea, do you think you could take a look? I suspect you'll know what the right fixes are right away.

**Comment by mordante** [comment]
From a quick look this looks like broken tests due to `TEST_HAS_NO_UNICODE` was never set. I'll create a commit with fixes.

File: libcxx/test/libcxx/utilities/format/format.functions/ascii.pass.cpp [hunk 1]
```diff
@@ -8,6 +8,7 @@
 // UNSUPPORTED: c++03, c++11, c++14, c++17
 // UNSUPPORTED: GCC-ALWAYS_INLINE-FIXME
 
+// What would be the best way to force the test with unicode disabled?
 // Force unicode to be disabled.
 // ADDITIONAL_COMPILE_FLAGS: -D_LIBCPP_HAS_NO_UNICODE
```
Comment (mordante) [inline libcxx/test/libcxx/utilities/format/format.functions/ascii.pass.cpp:13]: The approach by forcing the flags does not work after this patch; it probably should have never worked.
What would be the best approach, only define the macro when not defined yet, or mark this with
`REQUIRES: libcpp-has-no-unicode` ?

The same for the next file.

**Review by mordante — APPROVED** [review]
Still LGTM!

**Pull request merged**